### PR TITLE
Add support for JDBC Metadata for the nested typess List, Struct, Map

### PR DESCRIPTION
**Pull request by jonathanswenson**
These still don't work when you try to use JDBC to export the results. However it should be possible now to export arrow results directly.

Before, each of these types would fail when the query was executed even if you do not instantiate / use the JDBC ResultSet Metadata.

The exception would be: 
```
java.lang.IllegalArgumentException: No enum constant org.duckdb.DuckDBColumnType.BIGINT[]
	at java.base/java.lang.Enum.valueOf(Enum.java:273)
	at org.duckdb.DuckDBColumnType.valueOf(DuckDBColumnType.java:3)
	at org.duckdb.DuckDBResultSetMetaData.TypeNameToType(DuckDBResultSetMetaData.java:52)
	at org.duckdb.DuckDBResultSetMetaData.<init>(DuckDBResultSetMetaData.java:30)
	at org.duckdb.DuckDBNative.duckdb_jdbc_meta(Native Method)
	at org.duckdb.DuckDBPreparedStatement.prepare(DuckDBPreparedStatement.java:104)
	at org.duckdb.DuckDBPreparedStatement.executeQuery(DuckDBPreparedStatement.java:181)
```

### CODE REVIEW

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -495,6 +495,48 @@ public static void test_throw_wrong_datatype() throws Exception {
 		conn.close();
 	}
 
+	public static void test_list_metadata() throws Exception {
+		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
+		Statement stmt = conn.createStatement();
```
Comment (Mause) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:500]: Can you please add the missing close calls or wrap these in try-with-resources?

**Review by Mause — CHANGES_REQUESTED** [review]

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java (hunk 1 above)
Comment (jonathanswenson) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:500]: 👍🏻

Comment (jonathanswenson) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java]: Done thanks!

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 2]
```diff
@@ -495,6 +495,48 @@ public static void test_throw_wrong_datatype() throws Exception {
...
+					assertEquals(meta.getColumnName(1), "struct");
+					assertEquals(meta.getColumnTypeName(1), "STRUCT(i INTEGER, j VARCHAR)");
+					assertEquals(meta.getColumnType(1), Types.JAVA_OBJECT);
+				}
+			}
+		}
+	}
+
+	public static void test_map_metadata() throws Exception {
+		try (Connection conn = DriverManager.getConnection("jdbc:duckdb:")) {
```
Comment (Mause) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:527]: A bit of a nitpick, but you *can* collapse these three tries into one

Comment (jonathanswenson) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java]: 👍🏻

**Comment by jonathanswenson** [comment]
Technically fixes https://github.com/duckdb/duckdb/issues/2915 and https://github.com/duckdb/duckdb/issues/3687

Though I wouldn't be surprised if the original authors of these tickets actually wanted to extract these columns using the JDBC implementation rather than the arrow impl.

In other news, the array, map, and struct data can be extracted correctly with the arrow implementation.

**Comment by jonathanswenson** [comment]
@Mause would you like to create an issue specifically for this? As the other two tickets referenced above are somewhat different? Or file issues / Feature Requests specifically for the lack of support for complex types in the JDBC driver?

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**